### PR TITLE
fix: FluentMessageBar onclick bug and not using Link?.Target

### PR DESCRIPTION
**Pull request by benniedejong**
In the FluentMessageBar some @onclick event bindings were broken since version 4.4.0, I replaced them with the new OnClick. If a Target was specified in the Link of a notification, it was not used in the FluentMessageBar and always fell back to _blank.

# Pull Request

## 📖 Description
1. Fix: When opening the FluentMessageBar, it would create the unhandledexception "Unhandled exception rendering component: Unable to set property 'onclick' on object of type 'Microsoft.FluentUI.AspNetCore.Components.FluentAnchor'. The error was: Arg_InvalidCastException", this has been fixed by replacing the @onclick with OnClick

2. If the Link in the Message would contain a target, this target was not used (hardcoded to _blank). Now it user the target in the link or _blank in case no target has been specified.

## 👩‍💻 Reviewer Notes
Add messages to the FluentMessageBar and open it.

### General

- [ ] I have added tests for my changes.
- ✅ I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ ] I have read the [CONTRIBUTING](https://github.com/Microsoft/fluentui-blazor/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have added [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for my new compontent
- ✅ I have modified an existing component
- [ ] I have validate [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for an existing component

### CODE REVIEW

**Comment by benniedejong** [comment]
> @microsoft-github-policy-service agree

@microsoft-github-policy-service agree

File: src/Core/Components/MessageBar/FluentMessageBar.razor [hunk 1]
```diff
@@ -107,16 +107,16 @@
...
+                                  OnClick="@LinkClickedAsync"
                                   title="Link">
                         @Link?.Text
                     </FluentAnchor>
                 }
                 @if (ShowPrimaryAction)
                 {
-                    <FluentButton Appearance="Appearance.Hypertext"
-                                  @onclick="@PrimaryActionClickedAsync"
+                    <FluentButton Appearance="Appearance.Neutral"
```
Comment (dvoituron) [inline src/Core/Components/MessageBar/FluentMessageBar.razor:118]: Why did you changed the `Appearance`?

**Review by dvoituron — CHANGES_REQUESTED** [review]

**Comment by dvoituron** [comment]
Could you also solve the same `<FluentAnchor @onclick=""` problem for other components?
- NotificationCenterPanel.razor
- FluentToast.razor

File: src/Core/Components/MessageBar/FluentMessageBar.razor (hunk 1 above)
Comment (benniedejong) [reply src/Core/Components/MessageBar/FluentMessageBar.razor:118]: The FluentButton throws an ArgumentException when Appearance.Hypertext is used (FluentButton.razor.cs line 174, "Appearance.Hypertext is not supported for FluentButton").  I did see that other FluentButtons (for exmaple FluentMessageBar line 44) also use Neutral.

**Comment by benniedejong** [comment]
> Could you also solve the same `<FluentAnchor @onclick=""` problem for other components?
> 
> * NotificationCenterPanel.razor
> * FluentToast.razor

Will do FluentToast.razor tonight, NotificationCenterPanel.razor is already in my commit.

File: src/Core/Components/MessageBar/FluentMessageBar.razor (hunk 1 above)
Comment (dvoituron) [reply src/Core/Components/MessageBar/FluentMessageBar.razor:118]: OK. Makes sense. The design won't be the same in the notification center, but why not.

**Comment by vnbaaij** [comment]
The fix for FluentToast is already taken care of in an upcoming PR from me. And NotificationCenterPanel as well, so that one will be double fixed 😉

**Review by vnbaaij — APPROVED** [review]

**Pull request merged**